pkg/controller/operators/catalog/operator.go

```diff
@@ -1861,6 +1863,20 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 				gvk := unstructuredObject.GroupVersionKind()
 				r, err := o.apiresourceFromGVK(gvk)
 				if err != nil {
+					// Check if the API for this dynamic resource is available on cluster before creating the resource by checking discovery.
+					// Ensures CRs created via the dynamic client are created after the corresponding CRD is accepted and established on the cluster to avoid potential race conditions.
+					if _, ok := err.(olmerrors.GroupVersionKindNotFoundError); ok {
+						// API not yet established on cluster
+						o.logger.Infof("failed to find server resource for GVK %s via discovery", gvk.String())
+						plan.Status.Plan[i].Status = v1alpha1.StepStatusUnknown
+						continue
+					}
+					if k8serrors.IsNotFound(err) {
```

Comment:
Yeah, I agree -- originally I just wrote my own logic for this check, then scrolled down and saw this function and decided to reuse existing code (to limit the complexity of the PR). But I agree having it return two error types is potentially confusing. 

Yep there's another function that does the same thing haha -- it's just in the olm operator and not in the catalog operator. I'd prefer not to touch ` isGVKRegistered` and look into refactoring `apiresourceFromGVK` after looking at its usages as you suggested. Although since they are essentially the same function I'm wondering if it's worth consolidating them both as part of this PR?

Reply:
I'd like to see at least `apiresourceFromGVK` refactored, because it's directly impacting the way this change is implemented. There don't seem to be too many uses of its twin, so it might be an easy win to unify them now, but that's more of a tangent and I don't feel as strongly about it.

Reply:
Yep, sounds good -- I looked and the only actual use of `apiresourceFromGVK` function seems to be in a unit test -- and it doesn't make use of the custom error type at all. So my plan is to simply remove the return of the custom error (change it to a simple `fmt.Errorf()` error) in `apiresourceFromGVK` and have the simpler `if err != nil` condition in my code only

Reply:
`isGVKRegistered` also doesn't seem to actually require the use of the custom-error type at any of its call-sites, but I agree changing it is not entirely within the scope of this PR